Jenkinsfile

```diff
@@ -17,18 +17,60 @@ pipeline {
                         error 'Only develop branch can be merged into master'
                     }
                 }
-                sh 'gradle build -x test'
+                sh 'echo "Warming up"'
+                sh 'gradle -q'
+            }
+        }
+        stage('Build') {
+            parallel {
+                stage('Build Back-end') {
+                    steps {
+                        sh 'gradle build -x test -x bundleClient'
+                    }
+                }
+                stage('Build Front-end') {
+                    steps {
+                        sh 'gradle pnpmBuild'
+                    }
+                }
             }
         }
         stage('Test') {
+            parallel {
+                stage('Test Back-end') {
+                    steps {
+                        sh 'gradle test'
```

Comment:
can we exclude the build phase ? technically it will be skipped thanks to the cache, but there's still non negligible a check.